File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/controller/book/ReturnBookController.java
package dev.starryeye.book_library.controller.book;

import dev.starryeye.book_library.application.facade.ReturnBookFacade;
import dev.starryeye.book_library.controller.book.request.ReturnBookRequest;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class ReturnBookController {

    private final ReturnBookFacade facade;

    @PostMapping("/api/v1/books/return")
    public void returnBook(@Valid @RequestBody ReturnBookRequest request) {
        facade.returnBook(request.toInput());
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/controller/book/ReturnBookController.kt
package dev.starryeye.book_library.controller.book

import dev.starryeye.book_library.application.facade.ReturnBookFacade
import dev.starryeye.book_library.controller.book.request.ReturnBookRequest
import dev.starryeye.book_library.controller.book.request.toInput
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class ReturnBookController(
    private val facade: ReturnBookFacade
) {

    @PostMapping("/api/v1/books/return")
    fun returnBook(
        @Valid @RequestBody request: ReturnBookRequest,
    ) {
        facade.returnBook(request.toInput())
    }
}